providers/oidc_test.go

```diff
@@ -188,6 +192,43 @@ func TestOIDCProviderRedeem_custom_userid(t *testing.T) {
 	assert.Equal(t, defaultIDToken.Phone, session.Email)
 }
 
+func TestOIDCProviderRedeem_group_claim_exists(t *testing.T) {
```

Comment:
Now that there are more cases we are handling for `TestOIDCProviderRedeem` -- does it make sense to merge all of them into 1 function and use a test table?

I think a test table likely didn't make sense when `TestOIDCProviderRedeem_custom_userid` was the only addition a couple months back, but now that we have more permutations of inputs, one might make sense.

Reply:
I implemented most of the other suggested changes, but will hold off on this until we see where #466 is going and if that will end up merged instead.

Reply:
Can we take a look at getting these together under 1 `TestOIDCProviderRedeem` function using a test table for the use cases?